src/test/java/com/google/devtools/build/lib/rules/cpp/SkylarkCcCommonTest.java

```diff
@@ -1275,6 +1275,76 @@ public void testIncompatibleDepsetForLibrariesToLinkGetter() throws Exception {
         .containsExactly("a.so", "liba_Slibdep2.so", "b.so", "e.so", "liba_Slibdep1.so");
   }
 
+  @Test
+  public void testSolibLinkDefault() throws Exception {
+    setUpCcLinkingContextTest();
+    scratch.file(
+        "foo/BUILD",
+        "load('//tools/build_defs/cc:rule.bzl', 'crule')",
+        "cc_binary(name='bin',",
+        "   deps = [':a'],",
+        ")",
+        "crule(name='a',",
+        "   dynamic_library = 'a.so',",
+        "   interface_library = 'a.ifso',",
+        ")");
+    ConfiguredTarget a = getConfiguredTarget("//foo:a");
+    RuleContext ruleContext = getRuleContext(a);
+    CcToolchainProvider toolchain =
+        CppHelper.getToolchain(
+            ruleContext, ruleContext.getPrerequisite("$cc_toolchain", Mode.TARGET));
+    StructImpl info = ((StructImpl) getMyInfoFromTarget(a).getValue("info"));
+    SkylarkNestedSet librariesToLink = info.getValue("libraries_to_link", SkylarkNestedSet.class);
+    assertThat(
+            librariesToLink.toCollection(LibraryToLink.class).stream()
+                .filter(x -> x.getDynamicLibrary() != null)
+                .map(x -> x.getDynamicLibrary().getRootRelativePath().relativeTo(toolchain.getSolibDirectory()).toString())
```

Comment:
nit: can you refactor x.getDynamicLibrary().getRootRelativePath().relativeTo(toolchain.getSolibDirectory()).toString() to a helper private static method, to make this slightly more readable? (the combination of chained calls and use of stream() makes this difficult to read)

Reply:
I've factored it out as `getSolibRelativePath`.